### PR TITLE
fix(Breadcrumb): add missing instance options and types

### PR DESCRIPTION
**Pull request by tkrugg**
**Summary**
#### 1. fix(Breadcrumb): add missing instance options, ref https://github.com/algolia/angular-instantsearch/issues/372
adds missing instance options to Breadcrumbs, namely
- `separator`
- `transformItems`

Full list of instance options is here: https://www.algolia.com/doc/api-reference/widgets/breadcrumb/js/?language=angular#connector-param-init-instance-options
fa9e67b

#### 2. fix(Breadcrumb): add missing `transformItems` story  

I would have added a story `separator` as well but my understanding is
implies using another dataset - not sure it's worth the trouble since
the separator is mostly just passed to the connector.

#### 3. fix(Breadcrumbs): add type to refine and createURL

**Result**

https://deploy-preview-515--angular-instantsearch.netlify.com/examples/storybook/?path=/story/breadcrumb--with-transformed-items

### CODE REVIEW

File: src/breadcrumb/breadcrumb.ts [hunk 1]
```diff
@@ -74,7 +77,7 @@ export class NgAisBreadcrumb extends BaseWidget {
   }
 
   public state: BreadcrumbState = {
-    createURL: noop,
+    createURL: () => '',
```
Comment (Haroenv) [inline src/breadcrumb/breadcrumb.ts]:
```suggestion
    createURL: () => '#',
```

🤔 probably this

File: src/breadcrumb/__tests__/breadcrumb.spec.ts [hunk 2]
```diff
@@ -39,10 +49,46 @@ describe('Breadcrumb', () => {
...
+      defaultState,
+      template: `
+        <ais-breadcrumb 
+          [attributes]="attributes"
+          rootPath="Audio" 
+          separator=" / " 
+          [transformItems]="transformItems">
+        </ais-breadcrumb>`,
+      TestedWidget: NgAisBreadcrumb,
+      methods: { transformItems, attributes },
```
Comment (samouss) [inline src/breadcrumb/__tests__/breadcrumb.spec.ts]: From what I've understood this parameter is used to expose values to the template correct? If I'm correct we should rename it to something more explicit. It tooks me some time to understand the purpose of the option.

Comment (Haroenv) [reply src/breadcrumb/__tests__/breadcrumb.spec.ts]: it's to add things to the prototype of the created class, and also used / similar in the stories. 

something like `additionalProperties` could _maybe_ work

Comment (tkrugg) [reply src/breadcrumb/__tests__/breadcrumb.spec.ts]: yes, I'll wait for all the PRs using it to be merged and bulk rename it.
It was originally copied from storybook helper and intended to pass methods.

Comment (tkrugg) [reply src/breadcrumb/__tests__/breadcrumb.spec.ts]: opened an issue for it https://github.com/algolia/angular-instantsearch/issues/580

**Pull request merged**